README.md

```diff
@@ -27,31 +27,167 @@ A simple Http server can be built using a few lines of code.
 
 ```scala
 import zio._
+
 import zio.http._
 
-object HelloWorld extends ZIOAppDefault {
+object RequestStreaming extends ZIOAppDefault {
 
   val app: HttpApp[Any] = 
     Routes(
       Method.GET / "text" -> handler(Response.text("Hello World!"))
     ).toHttpApp
 
-  override val run =
-    Server.serve(app).provide(Server.default)
+    // Creating HttpData from the stream
+    // This works for file of any size
+    val data = Body.fromStream(stream)
+
+    Response(body = data)
+  }
+
+  // Run it like any simple app
+  val run: UIO[ExitCode] =
+    Server.serve(app).provide(Server.default).exitCode
 }
 ```
 
-## Steps to run an example
+ZIO-HTTP provides a core library, `zhttp-http`, which includes the base HTTP implementation and server/client capabilities based on ZIO. Additional functionality like serverless support, templating, and websockets are available through separate add-on modules. ZIO-HTTP applications can be easily integrated into different deployment platforms, such as server-based, serverless, or compiled to native binaries.
+
+The principles of ZIO-HTTP are:
+
+- Application as a Function: HTTP services in ZIO-HTTP are composed of simple functions. The `HttpApp` type represents a function from an `HttpRequest` to a `ZIO` effect that produces an `HttpResponse`.
+- Immutability: Entities in ZIO-HTTP are immutable by default, promoting functional programming principles.
+- Symmetric: The same `HttpApp` interface is used for both defining HTTP services and making HTTP requests. This enables easy testing and integration of services without requiring an HTTP container.
+- Minimal Dependencies: The core `zhttp-http` module has minimal dependencies, and additional add-on modules only include dependencies required for specific functionality.
+- Testability: ZIO-HTTP supports easy in-memory and port-based testing of individual endpoints, applications, websockets/SSE, and complete suites of microservices.
+- Portability: ZIO-HTTP applications are portable across different deployment platforms, making them versatile and adaptable.
+
+By leveraging the power of ZIO and the simplicity of functional programming, ZIO-HTTP provides a robust and flexible toolkit for building scalable and composable HTTP services in Scala.
+
+## Quickstart
+
+Eager to start coding without delay? If you're in a hurry, you can follow the [quickstart](https://github.com/zio/zio-http/tree/main/zio-http-example) guide or explore the [examples repository](https://github.com/zio/zio-http/tree/main/zio-http-example), which demonstrates different use cases and features of ZIO-HTTP.
+
+## Module feature overview
```

Comment:
I am not a fan of creating content that closely mimics what is already available on http4k. Many of the contents under this section are not related to ZIO‌ HTTP.
I think it is better to focus on our strengths!

Reply:
hmm most of the stuff i worked on is in the docs directory..